index.ts

```diff
@@ -271,10 +281,55 @@ app.post("/quiz/:type/question/:questionId", (req, res) => {
     res.redirect(`/quiz/${typeOfQuiz}/question/${questionId + 1}`);
 })
 
-app.get("/quiz/:type/score", (req, res) => {
-    // e.g. http://localhost:3000/quiz/10rounds/score
-    res.render("score");
-})
+app.get("/quiz/:type/score", async (req, res) => {
+    const typeOfQuiz = req.params.type;
+    let scores: number[] = [];
+    let highScore: number;
+    
+    if (user === null) {
+        throw "User not found";
+    } else {
+        switch (typeOfQuiz) {
+            case "tenrounds":
+                scores = questions.map(q => {
+                    if (q.correct_character === q.answer_character && q.correct_movie === q.answer_movie) {
+                        return 1;
+                    } else if (q.correct_character === q.answer_character || q.correct_movie === q.answer_movie) {
+                        return 0.5;
+                    } else {
+                        return 0;
+                    }
+                });
+                highScore = user.highscore_tenrounds;
+                break;
+            case "suddendeath":
+                let consecutiveCorrect = 0;
```

Comment:
"**consecutiveCorrect**" variable doesn't seem to be used anywhere, so I'd delete it. You calculate the score into the "scores" array anyway, and afterwards in sumOfScores. So "consecutiveCorrect" is not needed.
(On line 309 as well)